framework/src/modules/chain/submodules/process_transactions.js

```diff
@@ -139,22 +140,73 @@ class ProcessTransactions {
 
 		const transactionsResponses = transactions.map(transaction => {
 			const transactionResponse = transaction.apply(stateStore);
+
 			roundInformation.apply(stateStore, transaction);
 			stateStore.transaction.add(transaction);
+
+			// We are overriding the status of transaction because it's from genesis block
+			transactionResponse.status = TransactionStatus.OK;
 			return transactionResponse;
 		});
 
-		const unappliableTransactionsResponse = transactionsResponses.filter(
-			transactionResponse => transactionResponse.status !== TransactionStatus.OK
+		return {
+			transactionsResponses,
+			stateStore,
+		};
+	}
+
+	// eslint-disable-next-line class-methods-use-this
+	async applyTransactions(transactions, tx = undefined) {
+		// Get data required for verifying transactions
+		const stateStore = new StateStore(library.storage, {
+			mutate: true,
+			tx,
+		});
+
+		await Promise.all(transactions.map(t => t.prepare(stateStore)));
+
+		// Verify total spending of per account accumulative
+		const transactionsResponseWithSpendingErrors = ProcessTransactions.verifyTotalSpending(
+			transactions,
+			stateStore
 		);
 
-		updateTransactionResponseForExceptionTransactions(
-			unappliableTransactionsResponse,
-			transactions
+		const transactionsWithoutSpendingErrors = transactions.filter(
+			transaction =>
+				!transactionsResponseWithSpendingErrors
+					.map(({ id }) => id)
+					.includes(transaction.id)
+		);
+
+		const transactionsResponses = transactionsWithoutSpendingErrors.map(
+			transaction => {
+				stateStore.account.createSnapshot();
+				const transactionResponse = transaction.apply(stateStore);
+				if (transactionResponse.status !== TransactionStatus.OK) {
```

Comment:
can you explain a bit why we have this logic of
```
if not ok -> do something
if ok -> do something
if not ok -> do something
```
this will have three checks each time, rather why not just do
```
if ok -> do something
else if (not ok) -> rest do here
```
?

Reply:
@ManuGowda Because if it's not ok in the first condition we are mutating the transactionResponse.status. It's very ugly and not ideal. However, that's the case.. For the moment we can ignore this.

Reply:
Bc of looking for exceptions being "do something" in this case: `if not ok -> updateTransactionResponseForExceptionTransactions`

Reply:
anyway, it was not convincing, if  `TransactionStatus.OK` is really --" "-- ok, then we don't need to look for exceptions. but as @yatki explained it's not something u can understand straightaway unless u have the context.